cpu/cc26x2_cc13x2/include/cc26x2_cc13x2_prcm.h

```diff
@@ -23,66 +23,186 @@
+#define DDI_0_OSC_CTL0_RCOSC_LF_TRIMMED         0x00001000
+#define DDI_0_OSC_CTL0_HPOSC_MODE_EN            0x00004000
+#define DDI_0_OSC_CTL0_CLK_DCDC_SRC_SEL_m       0x01000000
+#define DDI_0_OSC_CTL0_DOUBLER_RESET_DURATION   0x02000000
+#define DDI_0_OSC_CTL0_DOUBLER_START_DURATION   0x0C000000
+#define DDI_0_OSC_CTL0_BYPASS_RCOSC_LF_CLK_QUAL 0x10000000
+#define DDI_0_OSC_CTL0_BYPASS_XOSC_LF_CLK_QUAL  0x20000000
+#define DDI_0_OSC_CTL0_XTAL_IS_24M              0x80000000
+#define DDI_0_OSC_STAT0_PENDINGSCLKHFSWITCHING  0x00000001
+#define DDI_0_OSC_STAT0_SCLK_HF_SRC_m           0x10000000
+#define DDI_0_OSC_STAT0_SCLK_HF_SRC_s           28
+#define DDI_0_OSC_STAT0_SCLK_LF_SRC_m           0x60000000
+#define DDI_0_OSC_STAT0_SCLK_LF_SRC_s           29
 /** @} */
 
-/** @ingroup cpu_cc26x2_cc13x2_peripheral_memory_map
-  * @{
-  */
-#define DDI0_OSC_BASE       0x400CA000 /**< DDI0_OSC base address */
-/*@}*/
+/**
+ * @ingroup cpu_cc26x2_cc13x2_peripheral_memory_map
+ * @{
+ */
+#define DDI_DIR              0x00000000
+#define DDI_SET              0x00000080
+#define DDI_CLR              0x00000100
+#define DDI_MASK4B           0x00000200
+#define DDI_MASK8B           0x00000300
+#define DDI_MASK16B          0x00000400
+/**
+ * @brief   DDI0_OSC base address
+ */
+#define DDI0_OSC_BASE        (PERIPH_BASE + 0xCA000)
+/**
+ * @brief   DDI0_OSC 16-bit masked access base address
+ */
+#define DDI0_OSC_BASE_M16    (DDI0_OSC_BASE + DDI_MASK16B)
+/** @} */
+
+/**
+ * @brief   DDI_0_OSC register bank
+ */
+#define DDI_0_OSC            ((ddi0_osc_regs_t *) (DDI0_OSC_BASE))
+/**
+ * @brief   DDI_0_OSC 16-bit masked access register bank
+ */
+#define DDI_0_OSC_M16        ((ddi0_osc_regs_m16_t *) (DDI0_OSC_BASE_M16))
+
+/**
+ * @brief   Source clocks
+ * @{
+ */
+#define OSC_SRC_CLK_HF       0x00000001
+#define OSC_SRC_CLK_LF       0x00000004
+/** @} */
+
+/**
+ * @brief   Oscillators
+ * @{
+ */
+#define OSC_RCOSC_HF         0x00000000
+#define OSC_XOSC_HF          0x00000001
+#define OSC_RCOSC_LF         0x00000002
+#define OSC_XOSC_LF          0x00000003
+/** @} */
+
+/**
+ * @brief    DDI_0_OSC functions
+ * @{
+ */
+/**
+ * @brief   Get the source clock settings.
+ *
+ * @param[in] sclk Source clock to check.
+ *
+ * @return Returns the type of oscillator that drives the clock source.
+ */
+uint32_t osc_clock_source_get(uint32_t sclk);
 
-#define DDI_0_OSC ((ddi0_osc_regs_t *) (DDI0_OSC_BASE)) /**< DDI_0_OSC register bank */
+/**
+ * @brief   Configure the oscillator input to a source clock.
+ *
+ * Use this function to set the oscillator source for one or more of the system
+ * source clocks.
+ *
+ * @param[in] sclk Source clock.
+ * @param[in] osc  Oscillator to use.
+ */
+void osc_clock_source_set(uint32_t sclk, uint32_t osc);
+
+/**
+ * @brief   Check if the HF clock source is ready to be switched.
+ *
+ * If a request to switch the HF clock source has been made, this function can
+ * be used to check if the clock source is ready to be switched.
```

Comment:
How would such a request look like?

Reply:
A request to change SCLK_HF oscillator is made using `osc_clock_source_set(OSC_SRC_CLK_HF, osc)` where `osc` can be either the internal RC oscillator or the external crystal oscillator.

I've added a little bit more documentation to explain how it's done.

Reply:
Thanks for clearing that up!
I suppose you have a particular use-case in mind that you aren't just making `osc_clock_source_set()` blocking?

Reply:
Yes, I actually request the change, perform some initialization on the RF Core and then I perform the switch of `SCLK_HF` to the crystal oscillator.

Although, it can be done in a blocking manner and would be more simple IMHO. What do you think?

Reply:
If you can do meaningful work while the clock is being re-configured go ahead - this is an internal API anyway, the only user is `cpu/cc26x2_cc13x`so we don't have to worry so much about consumers getting the usage right.

Then again, can't you just block in `osc_hf_source_switch()`? You'd have to block before anyway.

Reply:
I'll change it to a single blocking function to keeps things simple. Anyway this is just used on power-up/power-down of the RF Core. We can later check if it bring any benefits to switch the HF oscillator in a non-blocking way.